### ISSUE TITLE
The Citation-Key column disappears from time to time

### ISSUE THREAD

**Issue by ryofurue** [retorquere/zotero-better-bibtex#2249]
(I haven't been able to send a support log. See below.) My problem may be the same as

https://github.com/retorquere/zotero-better-bibtex/issues/2169

I'm not sure whether this report of mine is helpful or not, but . . . .

The Citation-Key column disappears from time to time.  It has been like this for a month or so. I haven't been able to identify a pattern about when that happens.

When it happens, I try to reinstate the column by right-clicking on the title row, but the "Citation Key" entry isn't there. (Not in the "More Columns" submenu, either.)  Restarting Zotero always fixes the problem.

So, I decided I would submit a report here when that happens next time.  It happened 10 minutes ago.  So, I enabled debug logging and went to "Send Better BibTeX support log . . . "   It said that it would send the generated log, but the "continue" button remained gray and unclickable.

macOS 12.5.1
Zotero 6.0.12
After restart, BBT was automatically upgraded to 6.7.21 .   I haven't experienced the problem with this version.

**Comment by retorquere** [comment]
There is a `Save copy of log` button that will generate a zipfile, you can attach that here. If the column does not appear, that means there's an initialization error, and I need the log for that.

Alternately, you can install https://github.com/retorquere/zotero-debug-log . This plugin doesn't affect Zotero in any way and gives a separate help-menu entry to submit a log. It's an emergency fallback for the support log system. It'd be helpful to have a log from that when the regular log submission does not work, there must be an error that's breaking the code that enables the `continue` button.

**Comment by ryofurue** [comment]
@retorquere Thank you for your help. I've installed the zotero-debug-log plugin and I'll follow your instructions when the problem happens next time.

**Comment by ryofurue** [comment]
JHTF6QNT-apse

Zotero has just lost the citekey column. Right-click on the title row doesn't include "citekey" option.

By the way, "Help" > "Send debug log to file.io" resulted in the following error:

````
[Exception... "Component not initialized"  nsresult: "0xc1f30001 (NS_ERROR_NOT_INITIALIZED)"  location: "JS frame :: chrome://zotero-debug-log/content/zotero-debug-log.js :: node_modules/jszip/dist/jszip.min.js/</<[54]</e.exports< :: line 3071"  data: no]
````

**Comment by retorquere** [comment]
The current log has started debug logging after the problem occurred, so no clues. I've published a new version of debug-log which will force debug logging on from the start. Please upgrade that, and alas, we will have to wait until it happens again after that.

**Comment by ryofurue** [comment]
KEHNN9QZ-apse

The same problem has finally happened. The above is the ID of the debug log: "Help" > "Send Better BibTeX debug log".  I sent the log as soon as I noticed the problem without doing anything else.

**Comment by retorquere** [comment]
The citation key field is added at startup, so I need a log that's running from startup. If you install https://github.com/retorquere/zotero-debug-log, Zotero will always start up with it enabled -- it might slow some things down, but the chances of getting a useful log are much increased.

**Comment by ryofurue** [comment]
> The citation key field is added at startup, so I need a log that's running from startup.

I thought the log I have sent you is that . . . On October 17, you published a new version of debug-log, which I installed right away.  The installation demanded a restart of Zotero . . . I think.  After that, have I restarted macOS once or twice?  Likely, but I don't remember.

Anyway, that' okay.  I'll send you another log if the problem hits again.

**Comment by retorquere** [comment]
You're right, sorry I missed that. The weird thing is that there is no error in the log, so it appears the column is added, even if it does not appear. I'll see what I can get from the Zotero devs on this.

**Comment by retorquere** [comment]
Can you try build 3439?

**Comment by ryofurue** [comment]
36HITG8F-apse

Now, my Zotero has lost its citation-key column. The above is the debug log I sent when I discovered it.  It's strange that I've never seen the "act of the crime" :-)

I'm not sure it's relevant but I've also sent the other debug log: D188937537

**Comment by retorquere** [comment]
I can't read the `D<number>` logs, those go to Zotero. Can it be the case that you have done something like the following:

* started Zotero, 
* closed the Zotero window with the red cross in the top-left but did not quit Zotero
* reopened the window using Window -> Zotero or Cmd-O

**Comment by ryofurue** [comment]
MMVFB32T-apse

> * started Zotero,
> * closed the Zotero window with the red cross in the top-left but did not quit Zotero
> * reopened the window using Window -> Zotero or Cmd-O

You've found a recipe to reproduce the problem 100% of the time!

The debug log at the top of this message was generated after I quit Zotero and followed your procedure (except that I reopened Zotero by clicking on the dock icon).

The remaining question (to myself) is why I had to close Zotero by the red button. I rarely do so because I don't like macOS's behavior of leaving apps open.  99% of the time, I just type ⌘Q to quit the app.

**Comment by ryofurue** [comment]
I've just noticed that the label has changed to "requires Zotero changes".  Does that mean a request has been submitted to the Zotero developers?

In any case, I've started to develop the habit of minimizing (instead of quitting or closing) Zotero when I want to temporarily remove its window.  So, this problem has stopped happening.

**Comment by retorquere** [comment]
I don't recall why I did that, so I'm going to assume it's my problem to fix.

**Comment by retorquere** [comment]
There isn't a fix for this in Z6, and in Z7 the problem is fixed. Z7 support is not far off, progress can be followed at #2522
